status-im/status-legacy
# Dialog added to ask permissions for webpage

fixes #TBD

### Summary
Followup on #11028
On privacy and security page new toggle added "Webview camera permissions request". 
When it DISABLED and some page wants camera access - user gets notification that access was blocked.
When it ENABLED and some page wants camera access - user asked if he wants to allow access.

<img width="331" alt="Screenshot 2020-09-10 at 08 52 40" src="https://user-images.githubusercontent.com/5786310/92686795-09bd9800-f343-11ea-9da8-7fd4a6289965.png">
<img width="331" alt="Screenshot 2020-09-10 at 08 52 26" src="https://user-images.githubusercontent.com/5786310/92686798-0c1ff200-f343-11ea-8538-4faa179c785b.png">

#### Platforms


* Android

#### Areas that maybe impacted
Web browsing

### Steps to test

* Test 1 
  * Open Status for a new account (`webview camera permissions request` toggle should be disabled)
  * open chat that contains link https://fatal0.netlify.com/android/webviewvideo.html 
  * click the link
  * make sure app notifies you that camera access request blocked

* Test 2
  * Go to `Privacy and Security` page, enable `webview camera permission request` toggle
  * open chat that contains link https://fatal0.netlify.com/android/webviewvideo.html 
  * click the link
  * make sure app asks if you want to deny/allow access request
  * press "Deny", make sure that picture from camera didn't appear on the page

* Test 3
  * open chat that contains link https://fatal0.netlify.com/android/webviewvideo.html 
  * click the link again
  * make sure app asks if you want to deny/allow access request
  * press "Allow", make sure that picture from camera did appear on the page. If camera access wasn't yet provided by system, you will have an additional system permission request.

status: wip
